feature/src/main/java/com/teamkkumul/feature/signup/SetProfileActivity.kt

```diff
@@ -71,23 +72,29 @@ class SetProfileActivity :
         setProfileViewModel.updateImageState.flowWithLifecycle(lifecycle).onEach {
             when (it) {
                 is UiState.Success -> {
-                    if (sourceFragment == SET_NAME_ACTIVITY) {
-                        inputName?.let { navigateToWelcome(it) }
-                    } else if (sourceFragment == MY_PAGE_FRAGMENT) {
-                        val intent = Intent().apply {
-                            putExtra(PROFILE_IMAGE_URL, selectedImageUri)
+                    when (sourceFragment) {
+                        SET_NAME_ACTIVITY -> {
+                            inputName?.let { navigateToWelcome(it) }
+                        }
+                        MY_PAGE_FRAGMENT -> {
+                            setProfileImageUri()
                         }
-                        setResult(Activity.RESULT_OK, intent)
-                        finish()
                     }
                 }
 
-                is UiState.Failure -> {}
                 else -> Unit
             }
         }.launchIn(lifecycleScope)
     }
 
+    private fun setProfileImageUri() {
+        val intent = Intent().apply {
+            putExtra(PROFILE_IMAGE_URL, selectedImageUri)
+        }
+        setResult(Activity.RESULT_OK, intent)
+        finish()
```

Comment:
```kotlin
Intent().apply {
    putExtra(PROFILE_IMAGE_URL, selectedImageUri)
}.also {
    setResult(Activity.RESULT_OK, it)
    finish()
}
``` 
이제 오비가 되셨으니 scope에 대한 코드리뷰 받으실까봐 미리 작성드립닏다! 이런식으로 쓰는 법도 있습니다만
아마 이런 코드 리뷰 많이 받으실 겁니다.

불필요한 변수를 선언함으로써 메모리 절약 -> 이건 잘못 된 이야기 입니다.
val을 선언하나 IntentI()객체로 묶나 동일하게 메모리 할당되며, 다만 참조를 위한 이름이 있다 없다 차이 입니다.
 
추후에 scope로 인한 불필요한 코드리뷰 받으실까봐 미리 말씀드립니다!

참고로 전 기존 코드를 사용할 것 같습니다!

Reply:
p1 : 여기서 image url을 넘길 필요는 없어보여요!
```suggestion
        setResult(Activity.RESULT_OK, Intent())
        finish()
```

Reply:
오 네엡-! 감사합니당 👍